explore/src/main/scala/explore/tabs/ObsTabTiles.scala

```diff
@@ -76,27 +79,31 @@ import java.time.Instant
 import scala.collection.immutable.SortedSet
 
 case class ObsTabTiles(
-  vault:                    Option[UserVault],
-  userId:                   Option[User.Id],
-  programId:                Program.Id,
-  modes:                    SpectroscopyModesMatrix,
-  backButton:               VdomNode,
-  observation:              UndoSetter[ObsSummary],
-  obsExecution:             Pot[Execution],
-  allTargets:               UndoSetter[TargetList],
-  allConstraintSets:        Set[ConstraintSet],
-  targetObservations:       Map[Target.Id, SortedSet[Observation.Id]],
-  focusedTarget:            Option[Target.Id],
-  searching:                View[Set[Target.Id]],
-  defaultLayouts:           LayoutsMap,
-  layouts:                  LayoutsMap,
-  resize:                   UseResizeDetectorReturn,
-  obsAttachments:           View[ObsAttachmentList],
-  obsAttachmentAssignments: ObsAttachmentAssignmentMap,
-  globalPreferences:        View[GlobalPreferences],
-  readonly:                 Boolean
+  vault:             Option[UserVault],
+  userId:            Option[User.Id],
+  programId:         Program.Id,
+  modes:             SpectroscopyModesMatrix,
+  backButton:        VdomNode,
+  observation:       UndoSetter[ObsSummary],
+  programSummaries:  UndoContext[ProgramSummaries],
+  focusedTarget:     Option[Target.Id],
+  searching:         View[Set[Target.Id]],
+  defaultLayouts:    LayoutsMap,
+  layouts:           LayoutsMap,
+  resize:            UseResizeDetectorReturn,
+  globalPreferences: View[GlobalPreferences],
+  readonly:          Boolean
 ) extends ReactFnProps(ObsTabTiles.component):
-  val obsId: Observation.Id = observation.get.id
+  val obsId: Observation.Id                                         = observation.get.id
+  val allConstraintSets: Set[ConstraintSet]                         = programSummaries.get.constraintGroups.map(_._2).toSet
```

Comment:
Nice, I guess theses extra params accumulated over time

Reply:
Part if it, I think, was trying to pass just what was needed. But, since we need to pass the ProgramSummaries now, might as well do it here.